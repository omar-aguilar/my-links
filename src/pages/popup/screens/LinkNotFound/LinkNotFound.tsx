import { useEffect, useState } from 'react';
import { useSearchParams } from 'react-router-dom';
import { parseShortLink } from '@/shared/utils';
import getHTTPSURLString from '@/shared/utils/getHTTPSURLString';
import useDocumentTitle from '../../hooks/useDocumentTitle';
import SearchResults from '../../components/SearchResults';
import AddShortLink from '../../components/AddShortLink';

const LinkNotFound = () => {
  useDocumentTitle('Link Not Found');
  const [searchParams] = useSearchParams();
  const [shortLink, setShortLink] = useState<string | null>(null);

  useEffect(() => {
    setShortLink(searchParams.get('shortLink'));
  }, [searchParams]);

  if (!shortLink) {
    return null;
  }

  const redirect = (shortLinkEntry: ShortLinkEntry) => {
    window.location.replace(getHTTPSURLString(shortLinkEntry.shortLink));
  };

  const { domain, slug, raw: initShortLink } = parseShortLink(shortLink);

  return (
    <>
      <h1>Link not found</h1>
      <AddShortLink onLinkAdded={redirect} initShortLink={initShortLink} />
      <SearchResults domain={domain} shortLink={slug} tag="" title="Maybe you meant" />
    </>
  );
};

export default LinkNotFound;
