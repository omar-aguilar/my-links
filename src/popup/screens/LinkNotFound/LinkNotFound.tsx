import { FunctionComponent, useEffect, useState } from 'react';
import { useSearchParams } from 'react-router-dom';
import useDocumentTitle from '../../hooks/useDocumentTitle';
import Similarities from '../../components/Similarities';
import AddLink from '../../components/AddLink/AddLink';
import styles from './LinkNotFound.scss';

const LinkNotFound: FunctionComponent = () => {
  useDocumentTitle('My Links - Link Not Found');
  const [searchParams] = useSearchParams();
  const [shortLink, setShortLink] = useState<string | null>(null);

  useEffect(() => {
    setShortLink(searchParams.get('shortLink'));
  }, [searchParams]);

  if (!shortLink) {
    return null;
  }

  return (
    <>
      <h1>
        <span>Link not found </span>
        <span className={styles['link-not-found']}>{shortLink}</span>
      </h1>
      <AddLink onLinkAdded={console.log} initShortLink={shortLink} />
      <Similarities shortLink={shortLink} />
    </>
  );
};

export default LinkNotFound;
