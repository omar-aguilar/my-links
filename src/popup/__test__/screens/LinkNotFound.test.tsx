import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import LinkNotFound from '../../screens/LinkNotFound';

describe('LinkNotFound', () => {
  it('renders without exploding', () => {
    const { container } = render(
      <MemoryRouter>
        <LinkNotFound />
      </MemoryRouter>
    );
    expect(container.ownerDocument.title).toEqual('My Links - Link Not Found');
    expect(container.firstChild).toMatchSnapshot();
  });
});
