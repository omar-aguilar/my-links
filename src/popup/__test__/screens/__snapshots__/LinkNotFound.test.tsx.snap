// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkNotFound renders without exploding 1`] = `null`;
